### PR TITLE
Generalized KeyTable select and removed export_* on VDS

### PR DESCRIPTION
**Pull request by jigold**
- Removed `export_samples`, `export_variants`, `export_genotypes`. To perform same functionality, need to convert to KeyTable and then use `select` and `export`.

- Changed Python interface for `select`, `drop`, and `key_by` to take varargs rather than a String or List of String.

- Select is not backwards compatible with 0.1. To select a column name with periods in it, must use backticks now.

- Not certain whether left hand side of named expression should be treated as an identifier or an annotationIdentifier. Right now, it's treated as an identifier. If it's an identifier, than `A.B = 5` will have a signature of `("A.B", TInt)`. If it's an annotationIdentifier path, than the signature would be `("A", TStruct(("B", TInt)))`. Thoughts?

### CODE REVIEW

File: src/main/scala/is/hail/expr/Parser.scala [hunk 1]
```diff
@@ -97,9 +104,33 @@ object Parser extends JavaTokenParsers {
     (types.toArray, () => fs.map(f => f()).toArray)
   }
 
+  def parseSelectExprs(codes: Array[String], ec: EvalContext, expectedHead: Option[String],
```
Comment (tpoterba) [inline src/main/scala/is/hail/expr/Parser.scala]: expectedHead seems to be unused

File: src/main/scala/is/hail/expr/Parser.scala [hunk 2]
```diff
@@ -97,9 +104,33 @@ object Parser extends JavaTokenParsers {
...
+    mangle: Boolean = false): (Array[List[String]], Array[Type], () => Array[Any], Array[Boolean]) = {
+    val idPaths = codes.map { x => select_arg.parse_opt(x) }
+    val (maybeNames, types, f, isNamed) = parseNamedExprs[List[String]](codes.mkString(","), annotationIdentifier,
+      ec, (t, s) => t.map(_ :+ s), Some((i) => idPaths(i)))
+
+    if (maybeNames.exists(_.isEmpty))
+      fatal("left-hand side required in annotation expression")
+
+    val names = maybeNames.map(_.get)
```
Comment (tpoterba) [inline src/main/scala/is/hail/expr/Parser.scala]: no .get!

**Comment by jigold** [comment]
Reverted the interface for `select`, `drop`, and `key_by` to requiring a str or list of str rather than varargs. This is because having default values specified by a kwarg was interacting poorly with the varargs. Python3 supports this better. https://stackoverflow.com/questions/13821877/function-call-with-named-unnamed-and-variable-arguments-in-python

File: python/hail/dataset.py [hunk 3]
```diff
@@ -356,7 +356,9 @@ def annotate_samples_expr(self, expr):
         Compute per-sample GQ statistics for hets:
 
         >>> vds_result = (vds.annotate_samples_expr('sa.gqHetStats = gs.filter(g => g.isHet()).map(g => g.gq).stats()')
-        ...     .export_samples('output/samples.txt', 'sample = s, het_gq_mean = sa.gqHetStats.mean'))
+        ...                  .samples_table()
+        ...                  .select('sample = s', 'het_gq_mean = sa.gqHetStats.mean')
```
Comment (catoverdrive) [inline python/hail/dataset.py]: Does this need to get changed back to .select(['sample = s', 'het_gq_mean = sa.gqHetStats.mean']) for the vararg/kwarg fix? Docs are failing here.

File: python/hail/dataset.py [hunk 4]
```diff
@@ -3210,7 +3043,9 @@ def lmmreg(self, kinshipMatrix, y, covariates=[], global_root="global.lmmreg", v
 
         The simplest way to export all resulting annotations is:
 
-        >>> lmm_vds.export_variants('output/lmmreg.tsv.bgz', 'variant = v, va.lmmreg.*')
+        >>> lmm_vds.variants_table()
+        ...        .select('variant = v', 'va.lmmreg.*')
```
Comment (catoverdrive) [inline python/hail/dataset.py]: same thing here

File: python/hail/dataset.py [hunk 5]
```diff
@@ -5376,7 +5211,9 @@ def tdt(self, pedigree, root='va.tdt'):
 
         >>> pedigree = Pedigree.read('data/trios.fam')
         >>> (vds.tdt(pedigree)
-        ...     .export_variants("output/tdt_results.tsv", "Variant = v, va.tdt.*"))
+        ...     .variants_table()
+        ...     .select('Variant = v', 'va.tdt.*')
```
Comment (catoverdrive) [inline python/hail/dataset.py]: and here

**Pull request merged**